### PR TITLE
Bump prometheus-operator to 0.40.0 version

### PR DESCRIPTION
**Pull request by lilic**
I believe this should fix the errors we have been seeing when bumping to 0.40 prometheus-operator, this is due the fact we were using the apiVersion: apiextensions.k8s.io/v1beta1 and this switches to using apiVersion: apiextensions.k8s.io/v1.

cc @openshift/openshift-team-monitoring 

* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by lilic** [comment]
> level=error msg="Error: Error launching source instance: InsufficientInstanceCapacity: We currently do not have sufficient m5.xlarge capacity in the Availability Zone you requested (us-east-1b). Our system will be working on provisioning additional capacity. You can currently get m5.xlarge capacity by not specifying an Availability Zone in your request or choosing us-east-1a, us-east-1c, us-east-1d, us-east-1f."

Seems unrelated.

/retest

**Comment by lilic** [comment]
/hold

Debugging locally, seems like there is another permissions missing.

**Comment by lilic** [comment]
Typo, should be fixed now. :)

File: jsonnet/prometheus-operator-user-workload.jsonnet [hunk 1]
```diff
@@ -18,6 +18,40 @@ local tlsVolumeName = 'prometheus-operator-user-workload-tls';
     '0thanosrulerCustomResourceDefinition':: {},
 
     clusterRole+: {
+     rules+:
+      [
```
Comment (simonpasquier) [inline jsonnet/prometheus-operator-user-workload.jsonnet:22]: (nit) can you add a TODO so we don't forget to remove these lines?

File: jsonnet/prometheus-operator-user-workload.jsonnet [hunk 2]
```diff
@@ -18,6 +18,40 @@ local tlsVolumeName = 'prometheus-operator-user-workload-tls';
     '0thanosrulerCustomResourceDefinition':: {},
 
     clusterRole+: {
+     roles+:
```
[the remark this answers is not in the document]
Comment (lilic) [reply jsonnet/prometheus-operator-user-workload.jsonnet]: I mean I will remove them this will hopefully if 0.40 gets merged this week, but can do.

**Comment by lilic** [comment]
> --- FAIL: TestAlertmanagerKubeRbacProxy (15.68s)

Was the only failure here. Retesting.

/retest

**Comment by lilic** [comment]
aws failure was networking and mongodb 🤷‍♀️ related.
/retest

**Comment by lilic** [comment]
> level=error msg="Error: Error launching source instance: InsufficientInstanceCapacity: We currently do not have sufficient m5.xlarge capacity in the Availability Zone you requested (us-east-1b). Our system will be working on provisioning additional capacity. You can currently get m5.xlarge capacity by not specifying an Availability Zone in your request or choosing us-east-1a, us-east-1c, us-east-1d, us-east-1f."

Hit the aws InsufficientInstanceCapacity again...

/retest

**Comment by lilic** [comment]
> level=error
level=error msg="Error: Error launching source instance: InvalidNetworkInterface.InUse: Interface: [eni-00965ccfba58473bb] in use."
level=error msg="\tstatus code: 400, request id: ac8356bb-1ff3-4c51-99b3-39f45bd4ba78"
level=error
level=error msg="  on ../tmp/openshift-install-174798424/master/main.tf line 124, in resource \"aws_instance\" \"master\":"
level=error msg=" 124: resource \"aws_instance\" \"master\" {"
level=error
level=error

wow no luck today...
/retest

**Comment by lilic** [comment]
/retest

TestAlertmanagerKubeRbacProxy again it seems...

**Pull request merged**